iree/compiler/Codegen/Common/LinalgBufferizePass.cpp

```diff
@@ -575,6 +553,41 @@ static LogicalResult hasDestructiveUpdateLoopPattern(scf::ForOp forOp,
   return success();
 }
 
+/// Ties together operands for operand fusion as exists today by reusing buffer
+/// for the result for one of the inputs to do in-place update. Ideally we dont
+/// need to do this if the fusion just happens at vector level. To be removed
+/// when that is worked out and can be load-bearing. Conditions checked here are
+/// 1) the result does not use the value of the `outs` buffer.
+/// 2) the input has a single use (this op) and has the same indexing map as the
+///    result.
+/// 3) the input equivalence set does not have an interface binding, i.e. it is
+///    not using a buffer from the dispatch ABI.
+static void tieOperandsForOperandFusion(linalg::LinalgOp linalgOp,
+                                        BufferizationPlan &plan) {
+  for (auto result : enumerate(linalgOp.getOutputOperands())) {
+    if (linalgOp.payloadUsesValueFromOperand(result.value())) {
+      continue;
+    }
+    for (OpOperand *input : linalgOp.getInputTensorOperands()) {
+      if (input->get().hasOneUse() &&
+          (input->get().getType().cast<RankedTensorType>().getElementType() ==
+           result.value()
```

Comment:
Nit: use some local variables for stuff like `result.value()->get().getType().cast<...>()`? I typically find `if` checks spanning such a wide range difficult to read.